### ISSUE TITLE
Loading module at boot results in hang

### ISSUE THREAD

**Issue by kmoore134** [zfsonfreebsd/ZoF#2]
Within bhyve, booting our TrueOS snapshot install image. If we have the system auto-load the new ZFS module, it hard-locks at boot-up. Confirmed if I unload /boot/modules/zfs.ko and load manually after boot-up it works normally. 

Here's the image we're using to test:

https://builds.ixsystems.com/jenkins/job/TrueOS%20-%20Snapshot%20Build/

Here's the kernel output of bootup where it hangs:

```
Loading kernel...
/boot/kernel/kernel text=0x1631016 data=0x1d3268+0x819550 syms=[0x8+0x17f628+0x8+0x19c847]
Loading configured modules...
/boot/kernel/tmpfs.ko size 0xfa58 at 0x2709000
/boot/kernel/aesni.ko size 0xe2c0 at 0x2719000
/boot/kernel/geom_eli.ko size 0x1e398 at 0x2728000
can't find '/boot/entropy'
/boot/modules/zfs.ko size 0x825108 at 0x2747000
loading required module 'cryptodev'
/boot/kernel/cryptodev.ko size 0x7eb0 at 0x2f6d000
GDB: no debug ports present
KDB: debugger backends: ddb
KDB: current backend: ddb
---<<BOOT>>---
Copyright (c) 1992-2018 The FreeBSD Project.
Copyright (c) 1979, 1980, 1983, 1986, 1988, 1989, 1991, 1992, 1993, 1994
        The Regents of the University of California. All rights reserved.
FreeBSD is a registered trademark of The FreeBSD Foundation.
FreeBSD 13.0-CURRENT 329c23588(HEAD) GENERIC amd64
FreeBSD clang version 7.0.1 (tags/RELEASE_701/final 349250) (based on LLVM 7.0.1)
WARNING: WITNESS option enabled, expect reduced performance.
VT: init without driver.
CPU: AMD Ryzen 5 1600X Six-Core Processor            (3592.80-MHz K8-class CPU)
  Origin="AuthenticAMD"  Id=0x800f11  Family=0x17  Model=0x1  Stepping=1
  Features=0x1783fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,MMX,FXSR,SSE,SSE2,HTT>
  Features2=0xfed83203<SSE3,PCLMULQDQ,SSSE3,FMA,CX16,SSE4.1,SSE4.2,MOVBE,POPCNT,AESNI,XSAVE,OSXSAVE,AVX,F16C,RDRAND,HV>
  AMD Features=0x26500800<SYSCALL,NX,MMX+,FFXSR,Page1GB,LM>
  AMD Features2=0x40231fb<LAHF,CMP,ExtAPIC,CR8,ABM,SSE4A,MAS,Prefetch,SKINIT,WDT,TCE,DBE>
  Structured Extended Features=0x129<FSGSBASE,BMI1,AVX2,BMI2>
  XSAVE Features=0x1<XSAVEOPT>
  AMD Extended Feature Extensions ID EBX=0x1007<CLZERO,IRPerf,XSaveErPtr>
  TSC: P-state invariant
Hypervisor: Origin = "bhyve bhyve "
real memory  = 7465861120 (7120 MB)
avail memory = 6113779712 (5830 MB)
Event timer "LAPIC" quality 600
ACPI APIC Table: <BHYVE  BVMADT  >
arc4random: no preloaded entropy cache
ioapic0 <Version 1.1> irqs 0-31 on motherboard
Timecounter "TSC-low" frequency 1796399082 Hz quality 1000
random: entropy device external interface
[ath_hal] loaded
kbd1 at kbdmux0
module_register_init: MOD_LOAD (vesa, 0xffffffff810bd600, 0) error 19
random: registering fast source Intel Secure Key RNG
random: fast provider: "Intel Secure Key RNG"
000.000045 [4164] netmap_init               netmap: loaded module
nexus0
cryptosoft0: <software crypto> on motherboard
aesni0: <AES-CBC,AES-XTS,AES-GCM,AES-ICM,AES-CCM> on motherboard
acpi0: <BHYVE BVXSDT> on motherboard
acpi0: Power Button (fixed)
atrtc0: <AT realtime clock> port 0x70-0x71 irq 8 on acpi0
atrtc0: registered as a time-of-day clock, resolution 1.000000s
Event timer "RTC" frequency 32768 Hz quality 0
attimer0: <AT timer> port 0x40-0x43 irq 0 on acpi0
Timecounter "i8254" frequency 1193182 Hz quality 0
Event timer "i8254" frequency 1193182 Hz quality 100
hpet0: <High Precision Event Timer> iomem 0xfed00000-0xfed003ff on acpi0
Timecounter "HPET" frequency 16777216 Hz quality 950
Event timer "HPET" frequency 16777216 Hz quality 550
Event timer "HPET1" frequency 16777216 Hz quality 450
Event timer "HPET2" frequency 16777216 Hz quality 450
Event timer "HPET3" frequency 16777216 Hz quality 450
Event timer "HPET4" frequency 16777216 Hz quality 450
Event timer "HPET5" frequency 16777216 Hz quality 450
Event timer "HPET6" frequency 16777216 Hz quality 450
Event timer "HPET7" frequency 16777216 Hz quality 450
Timecounter "ACPI-fast" frequency 3579545 Hz quality 900
acpi_timer0: <32-bit timer at 3.579545MHz> port 0x408-0x40b on acpi0
pcib0: <ACPI Host-PCI bridge> port 0xcf8-0xcff on acpi0
pci0: <ACPI PCI bus> on pcib0
ahci0: <Intel ICH8 AHCI SATA controller> mem 0xc0000000-0xc00003ff irq 16 at device 3.0 on pci0
ahci0: AHCI v1.30 with 6 6Gbps ports, Port Multiplier not supported
ahcich0: <AHCI channel> at channel 0 on ahci0
ahci1: <Intel ICH8 AHCI SATA controller> mem 0xc0000400-0xc00007ff irq 17 at device 4.0 on pci0
ahci1: AHCI v1.30 with 6 6Gbps ports, Port Multiplier not supported
ahcich6: <AHCI channel> at channel 0 on ahci1
virtio_pci0: <VirtIO PCI Network adapter> port 0x2000-0x201f mem 0xc0002000-0xc0003fff irq 18 at device 5.0 on pci0
vtnet0: <VirtIO Networking Adapter> on virtio_pci0
vtnet0: Ethernet address: 58:9c:fc:01:6d:09
vtnet0: netmap queues/slots: TX 1/1024, RX 1/512
000.000158 [ 538] vtnet_netmap_attach       vtnet attached txq=1, txd=1024 rxq=1, rxd=512
isab0: <PCI-ISA bridge> at device 31.0 on pci0
isa0: <ISA bus> on isab0
atkbdc0: <Keyboard controller (i8042)> port 0x60,0x64 irq 1 on acpi0
atkbd0: <AT Keyboard> irq 1 on atkbdc0
kbd0 at atkbd0
atkbd0: [GIANT-LOCKED]
driver bug: Unable to set devclass (class: atkbdc devname: (unknown))
psm0: <PS/2 Mouse> irq 12 on atkbdc0
psm0: [GIANT-LOCKED]
psm0: model Generic PS/2 mouse, device ID 0
uart0: <16550 or compatible> port 0x3f8-0x3ff irq 4 flags 0x10 on acpi0
uart0: console (9600,n,8,1)
uart1: <16550 or compatible> port 0x2f8-0x2ff irq 3 on acpi0
vga0: <Generic ISA VGA> at port 0x3b0-0x3bb iomem 0xb0000-0xb7fff pnpid PNP0900 on isa0
can't handle raw ops yet!!!
can't handle raw ops yet!!!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
```

And here's the output when I load it by hand after disabling the module via the boot loader:

```
# kldload zfs.ko
can't handle raw ops yet!!!
can't handle raw ops yet!!!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
sysctl_warn_reuse: can't re-use a leaf (kstat.zfs.misc.dbufstats.cache_levels_bytes_N)!
can't handle raw ops yet!!!
ZFS filesystem version: 5
ZFS storage pool version: features support (5000)
can't handle raw ops yet!!!

```

**Comment by mattmacy** [comment]
Are you sure that you're not loading opensolaris?

**Comment by mattmacy** [comment]
Never mind, just tried it here and I see the same issue. Must be a sysinit running too early.

**Comment by mattmacy** [comment]
I can't kill my VM after doing that :-/ - am going to use this opportunity to update the host.

**Comment by mattmacy** [comment]
fixed in b917de909921062f226a9bc7d40172ef0d3963e5
